RealmSwift/Sync.swift

```diff
@@ -469,6 +469,97 @@ fileprivate extension SyncManagementObjectStatus {
     }
 }
 
+public extension SyncSession {
+    /// Whether a given notifier block will report the upload status or the download status of its session.
+    public enum ProgressDirection {
+        /// :nodoc:
+        case upload
+        /// :nodoc:
+        case download
+    }
+    /// How the notifier block will behave.
+    public enum ProgressMode {
+        /**
+         The block will be called forever, or until the user unregisters it.
+         The progress struct will always report the latest number of transferred bytes,
+         and the latest number of total bytes to be transferred.
+         */
+        case reportIndefinitely
+        /**
+         The block will, upon registration, store the total number of bytes
+         to be transferred. The progress struct will always report the latest number
+         of transferred bytes out of that original number of bytes to be
+         transferred. When the number of transferred bytes reaches or exceeds the
+         number of bytes to be transferred, the block will be automatically
+         unregistered.
+         */
+        case forCurrentlyOutstandingWork
+    }
+
+    /**
+     A token corresponding to a progress notification block.
+
+     Call `stop()` on the token to stop notifications. If the notification block has already
+     been automatically stopped, calling `stop()` does nothing. `stop()` should be called
+     before the token is destroyed.
+     */
+    public typealias NotificationToken = RLMProgressNotificationToken
+
+    /**
+     A struct encapsulating progress information, as well as useful helper methods.
+     */
+    public struct Progress {
+        /// The number of bytes that have been transferred.
+        public let bytesTransferred: UInt
+        /// The total number of bytes that have been transferred, and are pending transfer.
+        /// If the notifier is tracking downloads, this number represents the size of the
+        /// changesets generated by all other clients using the database.
+        /// If the notifier is tracking uploads, this number represents the size of the
+        /// changesets representing the local changes on this client.
+        public let bytesPending: UInt
+
+        /// The fraction of bytes transferred out of outstanding bytes to transfer.
+        public var fractionTransferred: Double {
+            let percentage = Double(bytesTransferred)/Double(bytesPending)
+            return percentage > 1 ? 1 : percentage
+        }
+
+        /// Whether all pending bytes have already been transferred.
+        public var isTransferComplete: Bool {
+            return bytesTransferred >= bytesPending
+        }
+
+        /// :nodoc:
+        fileprivate init(transferred: UInt, pending: UInt) {
+            bytesTransferred = transferred
+            bytesPending = pending
+        }
+    }
+
+    /**
+     Register a progress notifier to the session. This progress notifier allows your app to be
```

Comment:
s/notifier/notification block/?

"Register … to the session" should probably be "with the session".